app/controllers/coronavirus_form/postcode_lookup_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class CoronavirusForm::PostcodeLookupController < ApplicationController
+  def show
+    if session[:flash].present?
```

Comment:
When you first reviewed this we did have everything in one controller but that was mainly because there was only one route. In order to improve the format in which full addresses we displayed after selection we felt that we needed the extra route. Going back to a single controller way of doing things would help solve the issues that you mention above but would make that controller quite heavy with logic which we'd be keen to avoid. 

You'd also still be in the position where you'd have to redirect to the support address path in the event you get a 401 out of the OS API. We think it's best to show that error message on the support-address page so even if we were to do this at the very least we'd need to be storing that error message in the session.

Storing the error message in the session is a bit of an anti pattern but having the two controllers does cut down on controller logic and avoid us having a controller that was several hundred lines long. Happy to talk about this offline.